core/src/main/java/nl/nn/adapterframework/filesystem/ImapFileSystem.java

```diff
@@ -333,17 +334,17 @@ public void removeFolder(String folderName, boolean removeNonEmptyFolder) throws
 				for (int i = 0; i < mimeMultipart.getCount(); i++) {
 					MimeBodyPart bodyPart = (MimeBodyPart) mimeMultipart.getBodyPart(i);
 					if (bodyPart.getContentType().startsWith("text/html")) {
-						return new nl.nn.adapterframework.stream.Message(bodyPart.getInputStream());
+						return new PartMessage(bodyPart);
 					}
 				}
 				for (int i = 0; i < mimeMultipart.getCount(); i++) {
 					BodyPart bodyPart = mimeMultipart.getBodyPart(i);
 					if (bodyPart.getContentType().startsWith("text")) {
-						return new nl.nn.adapterframework.stream.Message(bodyPart.getInputStream());
+						return new PartMessage(bodyPart);
 					}
 				}
```

Comment:
Misschien mis ik iets maar volgens mij kan 1 for loop weg?

Reply:
Ik denk het niet. Nu krijg je het eerste bodyPart waar het type=text/html, en als dat er niet is het eerste part waar type=text